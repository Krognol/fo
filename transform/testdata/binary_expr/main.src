package main

type A[T] T

func main() {
	a0 := A[string]("foo")
	print(a0 + "bar")
	print(a0 == "bar")
	print(a0 < "bar")
	print(a0 > "bar")
	print(a0 != "bar")
	print(a0 <= "bar")
	print(a0 >= "bar")

	a1 := A[int](123)
	print(a1 + 456)
	print(a1 - 456)
	print(a1 * 456)
	print(a1 / 456)
	print(a1 % 456)
	print(a1 & 456)
	print(a1 | 456)
	print(a1 ^ 456)
	print(a1 << 456)
	print(a1 >> 456)
	print(a1 &^ 456)
	print(a1 == 456)
	print(a1 < 456)
	print(a1 > 456)
	print(a1 != 456)
	print(a1 <= 456)
	print(a1 >= 456)

	a2 := A[bool](true)
	print(a2 && false)
	print(a2 || false)
}
