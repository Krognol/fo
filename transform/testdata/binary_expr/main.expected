package main

type A interface{}

func main() {
	a0 := A("foo")
	print(a0.(string) + "bar")
	print(a0.(string) == "bar")
	print(a0.(string) < "bar")
	print(a0.(string) > "bar")
	print(a0.(string) != "bar")
	print(a0.(string) <= "bar")
	print(a0.(string) >= "bar")

	a1 := A(123)
	print(a1.(int) + 456)
	print(a1.(int) - 456)
	print(a1.(int) * 456)
	print(a1.(int) / 456)
	print(a1.(int) % 456)
	print(a1.(int) & 456)
	print(a1.(int) | 456)
	print(a1.(int) ^ 456)
	print(a1.(int) << 456)
	print(a1.(int) >> 456)
	print(a1.(int) &^ 456)
	print(a1.(int) == 456)
	print(a1.(int) < 456)
	print(a1.(int) > 456)
	print(a1.(int) != 456)
	print(a1.(int) <= 456)
	print(a1.(int) >= 456)

	a2 := A(true)
	print(a2.(bool) && false)
	print(a2.(bool) || false)
}
